src/views/Main/Visualizations/EntitySetVisualizationList.js

```diff
@@ -14,53 +16,119 @@ export class EntitySetVisualizationList extends React.Component {
       });
+    });
+  }
+
+  getPropertyTypes = (entityTypes) => {
+    const uniquePropertyTypeIds = new Set();
+    entityTypes.forEach((entityType) => {
+      entityType.properties.forEach((propertyTypeId) => {
+        uniquePropertyTypeIds.add(propertyTypeId);
+      });
+    });
+    return Promise.map((uniquePropertyTypeIds), (id) => {
+      return EntityDataModelApi.getPropertyType(id);
+    }).then((propertyTypes) => {
+      const idToPropertyType = {};
+      propertyTypes.forEach((propertyType) => {
+        idToPropertyType[propertyType.id] = propertyType;
+      });
+      return idToPropertyType;
+    });
+  }
+
+  checkWhetherToDisplayEntitySet = (entitySet, idToEntityType, idToPropertyType) => {
+    const accessChecks = [];
+    idToEntityType[entitySet.entityTypeId].properties.forEach((propertyId) => {
+      if (EdmConsts.EDM_NUMBER_TYPES.includes(idToPropertyType[propertyId].datatype)) {
+        accessChecks.push({
+          aclKey: [entitySet.id, propertyId],
+          permissions: [Permission.READ.name]
+        });
+      }
+    });
+
+    return AuthorizationApi.checkAuthorizations(accessChecks)
+    .then((authorizedProperties) => {
+      const visualizableProperties = authorizedProperties.filter((property) => {
+        return property.permissions.READ;
+      });
+      return (visualizableProperties.length > 1) ? entitySet : null;
+    });
+  }
+
+  loadVisualizableEntitySets = (entitySets, idToEntityType, idToPropertyType) => {
+    const entitySetPromises = entitySets.map((entitySet) => {
+      return this.checkWhetherToDisplayEntitySet(entitySet, idToEntityType, idToPropertyType);
+    });
+    Promise.all(entitySetPromises).then((allEntitySetsResults) => {
+      const visualizableEntitySets = allEntitySetsResults.filter((entitySet) => {
+        return entitySet;
+      });
+      this.setState({
+        entitySets: visualizableEntitySets,
+        asyncStatus: ASYNC_STATUS.SUCCESS
+      });
+    });
   }
 
   render() {
     const entitySetList = this.state.entitySets.map((entitySet) => {
-      const set = entitySet.entitySet;
       return (
         <button
-          onClick={() => {
-            this.props.displayEntitySetFn(set.name, set.type.namespace, set.type.name);
-          }}
-          className={styles.listItemButton}
-          key={set.name}
-        >
-          <div className={styles.entitySetName}>{set.name} ({set.type.namespace}.{set.type.name})</div>
-          <div className={styles.entitySetTitle}>{set.title}</div>
+            onClick={() => {
+              this.props.displayEntitySetFn(entitySet.id, entitySet.entityTypeId);
+            }}
+            className={styles.listItemButton}
+            key={entitySet.id} >
+          <div className={styles.entitySetTitle}>{entitySet.title}</div>
+          <div className={styles.entitySetFqn}>{entitySet.type.namespace}.{entitySet.type.name}</div>
         </button>
       );
     });
     return (
       <div>
```

Comment:
you don't have to wrap the `AsyncContent` in a `div`. I think you can just return `AsyncContent`